packages/one-app-bundler/webpack/loaders/common.js

```diff
@@ -39,30 +39,73 @@ const cssLoader = ({ name = '', importLoaders = 2 } = {}) => ({
-  const whitelistPatterns = configOptions.purgecss.whitelistPatterns
-    ? configOptions.purgecss.whitelistPatterns.map((pattern) => new RegExp(pattern, 'i'))
+// transform strings deep and greedy parameters into regex
+const reconcileSafeList = (safelist) => {
+  const configOptionsReconciled = safelist;
+
+  const greedy = safelist.greedy
+    ? safelist.greedy.map((pattern) => new RegExp(pattern, 'i'))
     : [];
-  const whitelistPatternsChildren = configOptions.purgecss.whitelistPatternsChildren
-    ? configOptions.purgecss.whitelistPatternsChildren.map((pattern) => new RegExp(pattern, 'i'))
+  configOptionsReconciled.greedy = greedy;
+  const deep = safelist.deep
+    ? safelist.deep.map((pattern) => new RegExp(pattern, 'i'))
     : [/:global$/];
-  if (configOptions.purgecss.disabled) return [];
+  configOptionsReconciled.deep = deep;
+
+  return configOptionsReconciled;
+};
+
+const purgeCssLoader = () => {
+  const { purgecss } = getConfigOptions();
+  if (purgecss.disabled) return [];
+
+  let { safelist } = purgecss;
+  if (purgecss.safelist && !Array.isArray(purgecss.safelist)) {
+    safelist = reconcileSafeList(purgecss.safelist);
+  }
+  let aggregatedStandard = [];
+  let safelistDeep = [/:global$/];
+  // aggregate the various whitelist options if safelist is not present
+  if (!safelist) {
+    if (purgecss.whitelistPatterns) {
+      console.warn('Purgecss: Using depreciated property whitelistPatterns');
+      aggregatedStandard = [
+        ...aggregatedStandard,
+        ...purgecss.whitelistPatterns.map((pattern) => new RegExp(pattern, 'i')),
+      ];
+    }
+    if (purgecss.whitelist) {
+      console.warn('Purgecss: Using depreciated property whitelist');
+      aggregatedStandard = [...aggregatedStandard, ...purgecss.whitelist];
+    }
+    if (purgecss.whitelistPatternsChildren) {
+      console.warn('Purgecss: Using depreciated property whitelistPatternsChildren');
+
+      safelistDeep = purgecss.whitelistPatternsChildren.map((pattern) => new RegExp(pattern, 'i'));
+    }
+  }
+  /* eslint-enable inclusive-language/use-inclusive-words --
+  re enable disabled */
   return [{
     loader: '@americanexpress/purgecss-loader',
     options: {
-      paths: [path.join(packageRoot, 'src/**/*.{js,jsx}'), ...configOptions.purgecss.paths || []],
-      extractors: configOptions.purgecss.extractors || [],
-      fontFace: configOptions.purgecss.fontFace || false,
-      keyframes: configOptions.purgecss.keyframes || false,
-      variables: configOptions.purgecss.variables || false,
-      whitelist: configOptions.purgecss.whitelist || [],
-      whitelistPatterns,
-      whitelistPatternsChildren,
+      paths: [path.join(packageRoot, 'src/**/*.{js,jsx}'), ...purgecss.paths || []],
+      extractors: purgecss.extractors || [],
+      fontFace: purgecss.fontFace || false,
+      keyframes: purgecss.keyframes || false,
+      variables: purgecss.keyframes || false,
+      safelist: safelist || {
```

Comment:
if `safelist` is falsy then `aggregatedStandard` and `safelistDeep` are being mutated (see https://github.com/americanexpress/one-app-cli/blob/64a12d1779bb3afd6b3b2b4130817636590a23f9/packages/one-app-bundler/webpack/loaders/common.js#L72-L84) but won't never be used because of `safelist ||`

Reply:
Well if safelist is falsy then 
``` 
safelist: safelist || {
        standard: aggregatedStandard,
        deep: safelistDeep,
        greedy: [],
        keyframes: purgecss.keyframes || false,
        variables: purgecss.keyframes || false,
     },
```
here we can expect the default object to be set for standard were we haeve aggregatedStandard and safelistDeep set